src/utils/TodosContext.tsx

```diff
@@ -0,0 +1,124 @@
+import React, { useState, useEffect, useContext } from 'react';
+
+import { getTodos } from '../api/todos';
+
+import { FilterParams } from '../types/FilterParams';
+import { ErrorMessages } from '../types/ErrorMessages';
+import { Todo } from '../types/Todo';
+import { USER_ID } from './user';
+import { TempTodo } from '../types/TempTodo';
+
+interface TodosContextInterface {
+  todos: Todo[],
+  setTodos: React.Dispatch<React.SetStateAction<Todo[]>>,
+
+  filteredTodos: Todo[],
+
+  errorMessage: ErrorMessages,
+  setErrorMessage: React.Dispatch<React.SetStateAction<ErrorMessages>>,
+
+  tempTodo: TempTodo | null,
+  setTempTodo: React.Dispatch<React.SetStateAction<TempTodo | null>>,
+
+  isAllCompleted: boolean | null,
+  setIsAllCompleted: React.Dispatch<React.SetStateAction<boolean | null>>,
+
+  filterParam: FilterParams,
+  setFilterParam: React.Dispatch<React.SetStateAction<FilterParams>>,
+
+  isCompletedTodosCleared: boolean,
+  setIsCompletedTodosCleared: React.Dispatch<React.SetStateAction<boolean>>,
+}
+
+export const TodosContext = React.createContext<TodosContextInterface>({
+  todos: [],
+  setTodos: () => {},
+
+  filteredTodos: [],
+
+  errorMessage: ErrorMessages.Default,
+  setErrorMessage: () => {},
+
+  tempTodo: null,
+  setTempTodo: () => {},
+
+  isAllCompleted: null,
+  setIsAllCompleted: () => {},
+
+  filterParam: FilterParams.All,
+  setFilterParam: () => {},
+
+  isCompletedTodosCleared: false,
+  setIsCompletedTodosCleared: () => {},
+});
+
+const getFilteredTodos = (todos: Todo[], completionStatus: FilterParams) => {
+  if (completionStatus === 'All') {
+    return [...todos];
+  }
+
+  const isCompleted = completionStatus === FilterParams.Completed;
+
+  return todos.filter(({ completed }) => completed === isCompleted);
+};
+
+type Props = {
+  children: React.ReactNode,
+};
+
+export const TodosProvider: React.FC<Props> = ({ children }) => {
+  const [errorMessage, setErrorMessage]
+  = useState<ErrorMessages>(ErrorMessages.Default);
+
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [tempTodo, setTempTodo] = useState<TempTodo | null>(null);
+  const [isAllCompleted, setIsAllCompleted]
+  = useState<null | boolean>(null);
```

Comment:
This state looks unnecessary, you can check isAllCompleted dynamically

Reply:
This state is used to change all todos (complete / uncomplete all at once)